core/src/services/resizeObserver.ts

```diff
@@ -1,45 +1,76 @@
 import type {ReadableSignal} from '@amadeus-it-group/tansu';
 import {derived} from '@amadeus-it-group/tansu';
-import {createBrowserStoreDirective} from '../utils/directive';
+import {createBrowserStoreArrayDirective} from '../utils/directive';
 import {noop} from '../utils/func';
 import type {Directive, SSRHTMLElement} from '../types';
 
 /**
- * Create a resize observer object
- * @returns An object containing the store with the dimentions of observed element (ResizeObserverEntry), the directive to be applied to the html element to be observed
+ * Create a resize observer that can be applied to multiple elements
+ * @returns An object containing the store with the dimensions map of observed elements (ResizeObserverEntry), the directive to be applied to the html element to be observed
  */
-export const createResizeObserver = (): {
-	dimensions$: ReadableSignal<ResizeObserverEntry | undefined>;
+export const createResizeObserverArray = (): {
+	dimensionsMap$: ReadableSignal<Map<HTMLElement, ResizeObserverEntry>>;
 	directive: Directive<void, SSRHTMLElement>;
 } => {
-	const {element$, directive} = createBrowserStoreDirective();
+	const {elements$, directive} = createBrowserStoreArrayDirective();
 
-	const observedElement$ = derived<ResizeObserverEntry | undefined, ReadableSignal<HTMLElement | null>>(
-		element$,
-		(element, set) => {
-			if (element === null) {
+	const dimensionsMap$ = derived(
+		elements$,
+		(elements, set) => {
+			const dimensionsMap = new Map<HTMLElement, ResizeObserverEntry>();
+
+			if (elements.length === 0) {
+				set(dimensionsMap);
 				return noop;
 			}
 
 			const observer = new ResizeObserver((entries) => {
-				set(entries[0]);
+				entries.forEach((entry) => dimensionsMap.set(entry.target as HTMLElement, entry));
+				set(dimensionsMap);
 			});
 
-			observer.observe(element);
+			elements.forEach((element) => observer.observe(element));
 
-			return () => observer?.disconnect();
+			return () => observer.disconnect();
 		},
-		undefined,
+		new Map(),
 	);
 
-	return {
-		/**
-		 * Store which contains the dimensions of the observed element (ResizeObserverEntry type)
-		 * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserverEntry | MDN documentation}
-		 */
-		dimensions$: observedElement$,
+	return {dimensionsMap$, directive};
+};
+
+/**
+ * Create a resize observer object for a single element
+ * @returns An object containing the store with the dimensions of observed element (ResizeObserverEntry), the directive to be applied to the html element to be observed
+ */
+export const createResizeObserver = (): {
+	dimensions$: ReadableSignal<ResizeObserverEntry | undefined>;
+	directive: Directive<void, SSRHTMLElement>;
+} => {
+	const {dimensionsMap$, directive: arrayDirective} = createResizeObserverArray();
+	let firstElement: HTMLElement | null = null;
+
+	const directive: Directive<void, SSRHTMLElement> = (element) => {
```

Comment:
```suggestion
	const directive: Directive<void, SSRHTMLElement> = browserDirective((element) => {
```
Then you can remove `as HTMLElement` line 55.